### PR TITLE
Change “privacy and reach” settings so that unchecking boxes always increase privacy and checking them always increase reach

### LINKED ISSUE #26493: Privacy and reach: unify settings so "all boxes checked" = "most discoverable/public"
## Pitch

On the `Privacy and reach` panel, invert the name and default for two existing settings so "all boxes checked" means "most discoverable/public"

| Old | Old setting name | … | New | New setting name
|------------------|---------------------------|----|-------------------|---------------------------
❌ | `Manually review new followers` | ➡ | ✔️  | `Automatically accept new followers`
❌ | `Hide follows and followers from profile` | ➡ | ✔️  | `Show follows and followers on profile`

**The resulting outcome remains the same as before.**  Accounts will still be unlocked by default (`locked` = `false`) and not hide followers/followed by (`hide_collections` = `false`).¹

This merely changes how it's presented to the user at the UI layer using [the new `setting_inverse_alias()` function introduced with the privacy tab](https://github.com/mastodon/mastodon/pull/26484 ).

```ruby
# Example addition to app/models/user_settings.rb
setting_inverse_alias :unlocked, :locked
setting_inverse_alias :show_collections, :hide_collections
```

New hint text:
* `Automatically accept new followers`
  * `People can follow you without requesting approval and you will not need to accept or reject new followers.`
* `Show follows and followers on profile`
  * `People can browse through your follows and followers.  People you follow will see that you follow them regardless.`

¹ Some folks pointed out wanting to change these defaults, or letting the instance administrator configure the defaults.  If so, I'd suggest doing that separately for easier review.

*[Feature request is sourced from my past comment](https://github.com/mastodon/mastodon/pull/26344#issuecomment-1676195299)*

## Motivation

* Makes it easier to opt in to the traditional social media experience ("just check all the boxes on this page!")
* Simplifies locking down one's account if being harassed ("uncheck all of the boxes on this page")
* Potentially reduces cognitive load by keeping a consistent privacy outcome (becoming more public/discoverable/reachable) for each checkbox

### Before
Screenshot of the "Privacy and reach" settings panel showing "Manually review new followers" and "Hide follows and followers from profile" unchecked
> * `Manually review new followers`
>   * `People will request to follow you and you will be able to accept or reject new followers.`
> * `Hide follows and followers from profile`
>   * `People will not be able to browse through your follows and followers.  People you follow will see that you follow them regardless.`

### After
Screenshot of the "Privacy and reach" settings panel showing "Automatically accept new followers" and "Show follows and followers on profile" checked and recommended
> * `Automatically accept new followers` (recommended)
>   * `People can follow you without requesting approval and you will not need to accept or reject new followers.`
> * `Show follows and followers on profile` (recommended)
>   * `People can browse through your follows and followers.  People you follow will see that you follow them regardless.`

### PR DESCRIPTION
**Pull request by ClearlyClaire**
Fixes #26493 (with some wording changes)

## Before

image

## After

image

### CODE REVIEW

File: app/models/account.rb [hunk 1]
```diff
@@ -442,8 +442,21 @@ def from_text(text)
...
+      end
+
+      define_method(key) do
+        !public_send(original_key)
+      end
+    end
   end
 
+  inverse_alias :show_collections, :hide_collections
+  inverse_alias :unlocked, :locked
```
Comment (ClearlyClaire) [inline app/models/account.rb:458]: I am not super happy with this bit, but it's the cleanest way I have found.

**Review by ykzts — APPROVED** [review]

**Comment by ClearlyClaire** [comment]
cc @digitalcircuit

**Review by digitalcircuit — APPROVED** [review]
Looks good to me, and thank you!

1.  To explicitly confirm since your screenshot shows different settings, the defaults for new accounts remain the same (profile unlocked, showing followers/followed), correct?

     * I don't feel strongly about this either way, I simply wasn't sure if it intentionally got changed at some point.

2.  I am unclear on what criteria Mastodon uses for the `Recommended` label and whether it fits, but I'm happy with this and it can always be tweaked later (e.g. when adding the public status indexing setting).

**Comment by vmstan** [comment]
It seems like anything that you're automatically opt-in on should be marked as a "Recommended" option.

Possibly outside the scope of this but a "Restore Defaults" button might also be helpful.

**Comment by ClearlyClaire** [comment]
> It seems like anything that you're automatically checked on should be marked as a "Recommended" option.

Historically, “Recommended” was introduced precisely because there was disagreement between what the default should be and what Gargron felt was the most useful (e.g. if I remember correctly, GIF autoplay is something we insisted was opt-in because of accessibility issues, but Gargron felt most people would want.

**Pull request merged**